Menus/BuyMenuGUI.cpp

```diff
@@ -1353,10 +1378,36 @@ void BuyMenuGUI::Update()
         }
 
         // Get handle to the currently selected item, if any
-        GUIListPanel::Item *pItem = m_pCartList->GetItem(m_ListItemIndex);
+        GUIListPanel::Item* pItem = m_pCartList->GetItem(m_ListItemIndex);
+        std::string description = "";
+
+        if (pItem && pItem->m_pEntity) {
+            description = pItem->m_pEntity->GetDescription();
+            const Entity* currentItem = pItem->m_pEntity;
+            std::stringstream floatStream;
+            const Actor* pItemAsActor = dynamic_cast<const Actor*>(currentItem);
+            if (pItemAsActor) {
+                floatStream << std::fixed << std::setprecision(1) << pItemAsActor->GetMass();
+                // If the item's mass is an integer, remove trailing zero and decimal point.
+                description += "\nMass: " + ((floatStream.str().back() == '0') ? floatStream.str().substr(0, floatStream.str().length() - 2) : floatStream.str());
+
+                int passengerSlotsTaken = pItemAsActor->GetPassengerSlots();
+                if (passengerSlotsTaken > 1) {
+                    description += "\nPassenger Slots: " + std::to_string(passengerSlotsTaken);
+                }
+            }
+            else {
+                const MovableObject* pItemAsMO = dynamic_cast<const MovableObject*>(currentItem);
+                if (pItemAsMO) {
+                    floatStream << std::fixed << std::setprecision(1) << pItemAsMO->GetMass();
+                    // If the item's mass is an integer, remove trailing zero and decimal point.
+                    description += "\nMass: " + ((floatStream.str().back() == '0') ? floatStream.str().substr(0, floatStream.str().length() - 2) : floatStream.str());
```

Comment:
Given how often you do this stringification, I think it'd be a good idea to add a method to RTETools that lets you pass in a number and get it stringified and trimmed/expanded to some passed-in precision. That way your code gets cleaner and other places have a consistent way of doing it that they can reuse.

Reply:
Now implemented as an external function, can round to variable precision. This'll require it's own review as well.